app.py

```diff
@@ -113,109 +16,109 @@ def llm_prompt(style, n_words):
-            $(document).on('keypress', function(e) {
-                if (e.which == 13) {  // 13 is the Enter key
-                    $('#go').click();
-                }
-            });
-        });
-        """)
+        ui.input_task_button("go", "Describe Image"),
     ),
-    ui.layout_columns(
-        ui.div(
-            ui.div(
-                ui.output_ui(
-                    "display_image",
-                    style="padding-bottom: 10px; display: flex; justify-content: center; align-items: flex-start; overflow: hidden;"
-                ),
-                style="flex-shrink: 1;"
-            ),
-            ui.div(
-                ui.output_ui("chat_container"),
-                style="flex-grow: 1; overflow-y: auto; display: flex; flex-direction: column-reverse;"
-            ),
-            style="display: flex; flex-direction: column; height: calc(100vh - 40px - 40px); justify-content: space-between;"
+    ui.layout_sidebar(
+        ui.sidebar(
+            ui.output_ui("chat_container"),
+            open="closed",
+            position="right",
+        ),
+        ui.layout_columns(
+            ui.output_ui("image_container", fill=True, fillable=True),
+            ui.output_ui("card_container", fill=True, fillable=True),
+            col_widths={"sm": (12), "lg": (6)},
         ),
-        ui.div(id = "card_container", style = "display: grid; grid-template-columns: 1fr;"),
-        col_widths={"sm": (12), "lg": (6)},
+        id="chat_sidebar",
+        fill=True,
+        padding="3rem",
+        border=False
     ),
-    title = "Image Describer",
+    ui.tags.style(".card-header p { margin-bottom: 0; }"),
+    ui.tags.script(src="keypress.js"),
+    fillable=True,
+    # TODO: padding doesn't work here?!
+    padding=0,
+    title="Image Describer",
 )
 
+
 def server(input, output, session):
     chat = ui.Chat("chat")
-    client = None
+    chat_client = ChatOpenAI(model="gpt-4o")
 
-    @output
-    @render.ui
-    def display_image():
-        return ui.img(src=input.url(), style="max-height: 50vh; display: flex; justify-content: center; align-items: center; overflow: hidden;")
+    card_title = ui.MarkdownStream("card_title")
+    card_body = ui.MarkdownStream("card_body")
 
-    def update_card(chunk, output):
-        output.append(chunk.content)
-        # try to update the card, but don't update if there's a parsing error
-        try:
-            card_update = parse_to_card("".join(output))
-            ui.insert_ui(card_update, "#card_container", immediate = True)
-            ui.remove_ui(selector = "#card_container > div:not(:last-child)", immediate = True)
-        except ValueError:
-            pass
-        return chunk.content
+    @reactive.effect
+    def _():
+        chat_client.system_prompt = llm_prompt(input.style(), input.n_words())
+
+    @render.ui
+    def image_container():
+        return ui.img(src=input.url(), class_="html-fill-item rounded")
```

Comment:
With this, I'm seeing the image stretched in the container. The following does work, but is there a better way?

```suggestion
        return ui.img(src=input.url(), class_="html-fill-item rounded", style="object-fit: scale-down; object-position: top;")
```

Reply:
Thanks, done in [9be0f39](https://github.com/jonkeane/shinyImages/pull/1/commits/9be0f394be224d8569d12c840268fbe5e14d8557)